### ISSUE TITLE
[BUG] Reagent Bag in bank not following rules

### ISSUE THREAD

**Issue by DravenSA** [arkayenro/arkinventory#1936]
Testing on the Alpha , noticed that any items in the Reagents bag, is no longer moving to a new bar according to a rule. 
is this a bug, or intended, or a setting that has changed?

Can you shed some light on the reagents bag not letting items out of the bag when a rule is made eg,

tt('Shadowlands Crafting Reagent')

Only some of the items will follow the rule as they were NOT inside the reagents bag, only once they were removed from the reagent bag in the bank to a normal bank bag,  did they follow the rule.

**Comment by arkayenro** [comment]
appears to be working ok for me.  are you sure you arent getting any errors?

**Comment by DravenSA** [comment]
Getting this when i login
```
3x ArkInventory/Core/ArkInventoryUtil.lua:17: ERROR> [string "return( not sb( ) and q( 2,3,4 ) and type (..."]:1: attempt to call global 'tsmvalue' (a nil value)
[string "=[C]"]: in function `error'
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:13: in function <ArkInventory/Core/ArkInventoryUtil.lua:11>
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:17: in function `Error'
[string "@ArkInventoryRules/ArkInventoryRules.lua"]:260: in function `AppliesToItem'
[string "@ArkInventory/Core/ArkInventoryCategory.lua"]:1614: in function `ItemCategoryGetPrimary'
[string "@ArkInventory/Core/ArkInventoryCategory.lua"]:1633: in function `ItemCategoryGet'
[string "@ArkInventory/Core/ArkInventoryAction.lua"]:151: in function `Check'
[string "@ArkInventory/Core/ArkInventory.lua"]:7645: in function `Frame_Item_Update_StatusIconJunk'
[string "@ArkInventory/Core/ArkInventory.lua"]:10061: in function `Frame_Item_Update_Instant'
[string "@ArkInventory/Core/ArkInventoryStorage.lua"]:555: in function <...aceArkInventory/Core/ArkInventoryStorage.lua:545>
[string "=[C]"]: ?
[string "@ACU/Libs/AceBucket-3.0-4/AceBucket-3.0.lua"]:69: in function <...faceACU/Libs/AceBucket-3.0/AceBucket-3.0.lua:67>
[string "@ACU/Libs/AceBucket-3.0-4/AceBucket-3.0.lua"]:83: in function `func'
[string "@ACU/Libs/AceTimer-3.0-17/AceTimer-3.0.lua"]:57: in function <ACU/Libs/AceTimer-3.0/AceTimer-3.0.lua:50>

Locals:
(*temporary) = "|cffff2020ERROR> [string "return( not sb( ) and q( 2,3,4 ) and type (..."]:1: attempt to call global 'tsmvalue' (a nil value)"

```

Here you can see that the Glimmer of Motion is in the reagent bag / tab of the bank and is not folloing the rule of the rest of them that were NOT in the reagent section

![Screenshot 2024-07-26 110825](https://github.com/user-attachments/assets/8936cd56-461e-4ec4-abb7-9addce862ba2)

**Comment by DravenSA** [comment]
I would also like to add that if i use an old Profile it looks like it is working, but it is not working when i create a new profile, 
So i have tried to copy from an existing profile, and again it wont work.

So i thought it might be a setting somewhere?

**Comment by DravenSA** [comment]
This is using an older profile and applying the same rule, and you can see that the same items are now highlighted and they are following the rule, so again this leads me to believe it has something to do with a  setting, or new profiles act differently?
![Screenshot 2024-07-26 111937](https://github.com/user-attachments/assets/6c1cc4a7-85ae-4fc2-989f-70f809aec7fe)

**Comment by arkayenro** [comment]
well thats weird, i found a profile where it didnt work, swapped back to one that did and when i swapped back to the one that wasnt working, it started working.

something must not be getting set in the profile properly but does if you swap around

**Comment by arkayenro** [comment]
nope, something much weirder is happening, now theyre stuck there even though the rule is disabled

**Comment by DravenSA** [comment]
Sorry to open a can or worms, but now at least i am not going crazy :-)

**Comment by arkayenro** [comment]
well that turned out to be a lot of code i screwed up.

should be fixed in the next alpha

**Comment by DravenSA** [comment]
Alpha 22 (but then i saw Alpha 23, and tried to update it on WoWUp, and it dropped me back to 21, and i saw on Curse that 23 was available, but i get an error trying to download it manually. )
Get errors as follows
```
33x CanIMogIt/Plugins/arkinventory.lua:6: attempt to index global 'ArkInventory' (a nil value)
[string "@CanIMogIt/Plugins/arkinventory.lua"]:6: in main chunk

Locals:
(*temporary) = nil
(*temporary) = "ArkInventory"
(*temporary) = true
(*temporary) = false
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index global 'ArkInventory' (a nil value)"
```

The Bank and bags dont load
And in the chat i get the following

![Screenshot 2024-07-26 144716](https://github.com/user-attachments/assets/bd4fa23c-9ca0-467e-a96e-c58c264fdefb)

**Comment by DravenSA** [comment]
Ok couldnt help myself, i saw 23 and then 24 on WowUp..... got 24 and still same errors as 22....... Seriously.... i will stop now till you give the go ahead :-)

**Comment by arkayenro** [comment]
download alpha 24, it should work, if it doesnt run these from the chat window

```
/run ArkInventory.EraseSavedData( nil, ArkInventory.Const.Location.Bag, false )
/run ArkInventory.EraseSavedData( nil, ArkInventory.Const.Location.Bank, false )
```

**Comment by arkayenro** [comment]
it should only screw up if you managed to download and run alpha 23 before i could delete it

**Comment by DravenSA** [comment]
Hi again, dont know if you want a new issue opened, but after a few hours of making rules and placing them in the new profile, i have noticed that i can no longer move bars to different places, eg drag bar 10 and move it to position 23.  checked old profile and the same thing cant move bars to new locations.

**Comment by DravenSA** [comment]
2 more errors,
```
35x ArkInventory/Core/ArkInventory.lua:6149: attempt to index local 'loc_id' (a number value)
[string "@ArkInventory/Core/ArkInventory.lua"]:6149: in function `Frame_Bar_Paint_All'
[string "@ArkInventory/Core/ArkInventoryMenu.lua"]:342: in function `editBoxFunc'
[string "@ArkInventory/Libs/ArkDewdrop/ArkDewdrop.lua"]:2379: in function <...eArkInventory/Libs/ArkDewdrop/ArkDewdrop.lua:2365>

Locals:
(for state) = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 4 = <table> {
 }
 8 = <table> {
 }
 16 = <table> {
 }
 17 = <table> {
 }
 9 = <table> {
 }
 18 = <table> {
 }
 19 = <table> {
 }
 10 = <table> {
 }
 20 = <table> {
 }
 21 = <table> {
 }
 22 = <table> {
 }
 3 = <table> {
 }
 6 = <table> {
 }
 12 = <table> {
 }
 13 = <table> {
 }
 7 = <table> {
 }
 14 = <table> {
 }
 15 = <table> {
 }
 5 = <table> {
 }
}
(for control) = 1
loc_id = 1
loc_data = <table> {
 canpreload = true
 isBeingScanned = false
 canView = true
 player_id = "Delish - Ahn'Qiraj"
 isOffline = false
 canCompress = true
 canOverride = true
 canSearch = true
 Texture = "Interface\Icons\INV_Misc_Bag_07_Green"
 preloaditeminfo = true
 retryCount = 0
 template = "ARKINV_TemplateButtonItem"
 id = 1
 changed = true
 drawState = 5
 maxBar = 110
 canRestack = true
 Name = "Bags"
 maxSlot = <table> {
 }
 hasChanger = true
 isMapped = true
 Internal = "bags"
}
(*temporary) = 1
(*temporary) = "ARKINV_Frame1ScrollContainer"
(*temporary) = "%s%s%s"
(*temporary) = "ARKINV_Frame"
(*temporary) = "1"
(*temporary) = "ScrollContainer"
(*temporary) = "ARKINV_Frame1ScrollContainer"
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'loc_id' (a number value)"
_G = <table> {
 ERR_OUT_OF_CHI = "Not enough chi"
 DH_HAVOC_CORE_ABILITY_2 = "Strong melee attack that consumes Fury. If it critical strikes, some Fury is refunded."
 UNIT_NAMES_COMBATLOG_TOOLTIP = "Color unit names."
 HealBot_Options_LoadSpellsbText = HealBot_Options_LoadSpellsbText {
 }
 LE_GAME_ERR_CHAT_RAID_RESTRICTED_TRIAL = 783
 SPELL_FAILED_CUSTOM_ERROR_71 = "This partygoer wants to dance with you."
 ClassTrialTimerDisplayMixin = <table> {
 }
 MultiBarLeftButton7 = MultiBarLeftButton7 {
 }
 MerchantItem7AltCurrencyFrameItem1Text = MerchantItem7AltCurrencyFrameItem1Text {
 }
 BtWQuestsFrameChainTooltipTexture2 = BtWQuestsFrameChainTooltipTexture2 {
 }
 OPTION_SHOW_ACTION_BAR5_TOOLTIP = "Attached to the left side of Action Bar 4 by default"
 BINDING_NAME_NAMEPLATES = "Show Enemy Nameplates"
 CHAT_HEADER_SUFFIX = ": "
 HealBot_Options_AutoShowButtonNormalTexture = HealBot_Options_AutoShowButtonNormalTexture {
 }
 COVENANT_MISSIONS_HEALTH = "Health"
 HealBot_Debug_FrameText31 = HealBot_Debug_FrameText31 {
 }
 PVPReadyDialogBottomArt = PVPReadyDialogBottomArt {
 }
 CALENDAR_RAID_RESET_DESCRIPTION = "%1$s resets at %2$s."
 RematchGameTooltipTexture13 = RematchGameTooltipTexture13 {
 }
 ChatFrame6EditBoxFocusMid = ChatFrame6EditBoxFocusMid {
 }
 AceConfigDialogTooltipTexture20 = AceConfigDialogTooltipTexture20 {
 }
 HealBot_Options_BackupBuffWellFedItemsButton = HealBot_Options_BackupBuffWellFedItemsButton {
 }
 LE_GAME_ERR_ONLY_ONE_QUIVER = 36
 INT_SPELL_DURATION_HOURS = "%d |4hour:hrs;"
 LOSS_OF_CONTROL_DISPLAY_FEAR = "Feared"
 IAT_Achievement_9_114 = IAT_Achievement_9_114 {
 }
 AllTheThings-Settings-Interface-Ad-Hoc Window Updates = AllTheThings-Settings-Interface-Ad-Hoc Window Updates {
 }
 Boss1TargetFrameValue = Boss1TargetFrameValue {
 }
 HEALBOT_OPTIONS_ALSO_WHEN_MOUNTED = "also when mounted"
 SoulbindCacheOpener_decaying-phlegmFreeSpaceFont = SoulbindCacheOpener_decaying-phlegmFreeSpaceFont {
 }
 ElvUI_Bar6Button6 = ElvUI_Bar6Button6 {
 }
 ACTION_SPELL_MISSED_POSSESSIVE = "1"
 RaidUtility_CloseButton = RaidUtility_CloseButton {
 }
 ARKINV_Frame4TitleActionButton2 = ARKINV_Frame4TitleActionButton2 {
 }
 SoulbindCacheOpener_phoenix-feather-quillCount = SoulbindCacheOpener_phoenix-feather-quillCount {
 }
 HealBot_Options_Bind4Icon = HealBot_Options_Bind4Icon {
 }
 LE_GAME_ERR_AZERITE_ESSENCE_SELECTION_FAILED_ESSENCE_NOT_UNLOCKED = 1074
 DUNGEON_FLOOR_DRAGONBLIGHTCHROMIESCENARIO2 = "Andorhal"
 L = <table> {
 }
 HealBot_Options_OverridePlayerTargetFramesText = HealBot_Options_OverridePlayerTargetFramesText {
 }
 WowTokenRedemptionFrameBg = WowTokenRedemptionFrameBg {
 }
 HEALBOT_ELIXIR_OF_POISON_RES = 3386
 MinArchRaceButton8 = MinArchRaceButton8 {
 }
 SPELL_FAILED_CUSTOM_ERROR_152 = "You must choose a specialization to use Ascendance."
 AllTheThings-Settings-General-
``` 

And this one was when i tried to right click the reagents bag in the bank to deposit all reagents, but the menu didnt open
```
2x ArkInventory/Core/ArkInventoryUtil.lua:235: ERROR> Map.Window[19] is not mapped
[string "=[C]"]: in function `error'
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:13: in function <ArkInventory/Core/ArkInventoryUtil.lua:11>
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:22: in function `Assert'
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:235: in function `MapGetWindow'
[string "@ArkInventory/Core/ArkInventoryUtil.lua"]:255: in function `getBlizzardBagIdFromWindowId'
[string "@ArkInventory/Core/ArkInventoryObject.lua"]:575: in function `ObjectStringDecode'
[string "@ArkInventory/Core/ArkInventoryObject.lua"]:929: in function `GetObjectInfo'
[string "@ArkInventory/Core/ArkInventoryMenu.lua"]:2277: in function `MenuBagOpen'
[string "@ArkInventory/Core/ArkInventory.lua"]:12176: in function `Frame_Changer_Slot_OnClick'
[string "*ArkInventory.xml:1141_OnClick"]:1: in function <[string "*ArkInventory.xml:1141_OnClick"]:1>

Locals:
(*temporary) = "|cffff2020ERROR> Map.Window[19] is not mapped"
```

**Comment by arkayenro** [comment]
both of those should be fixed in the next alpha

**Comment by arkayenro** [comment]
the edit mode issue with be fixed as well
